File: ack.swift
#!/usr/bin/xcrun swift
func ack(m: Int, n: Int) -> Int {
    if ( m == 0) {return n + 1}
    else if ( n == 0 ) {return ack(m - 1, n: 1)}
    else {return ack(m - 1, n: ack(m, n: n - 1))}
}

func main() {
    for i in 0..<5 {
        for j in 0..<5 {
            let ans: Int = ack(i, n: j)
            print("The ack of \(i) and \(j) is \(ans)")
            if i == 4 && j == 1 {
                break // Stack limit for Swift
            }
        }
    }
}

main()

File: ack.py
from __future__ import print_function
def ack(m, n):
    if ( m == 0 ):
        return n + 1
    elif ( n == 0 ):
        return ack(m - 1, 1)
    else:
        return ack(m - 1, ack(m, n - 1))

def main ():
    for i in range(5):
        for j in range(5):
            if i == 4 and j == 1: # Python Stack limit
                break
            ans = ack(i, j)
            print("The ack of %i and %i is %i" % (i, j, ans))
    return 0

if __name__ == '__main__':
    main()